frontend/src/components/PostField.jsx

```diff
@@ -1,22 +1,64 @@
-import React, { useState, Component } from "react";
+import React, { useState, useEffect } from "react";
 import "./style/PostField.scss";
 import axios from "axios";
 import { backend } from "../conf.js";
+import Postcard from "./Postcard";
 
 function PostField() {
   const user_id = 4;
   const [message, setMessage] = useState("");
+  const [file, setFile] = useState(null);
+  const [imagePreviewUrl, setImagePreviewUrl] = useState(null);
+  const [gamelist, setGamelist] = useState([]);
+  const [game_id, setGame_id] = useState(null);
 
-  const onSubmit = e => {
+  useEffect(() => {
+    axios.get(`${backend}/api/gamelist/`).then(({ data }) => {
+      setGamelist(data);
+    });
+  }, []);
+
+  const handleImageChange = e => {
     e.preventDefault();
-    const postObject = {
-      user_id,
-      message
+
+    let reader = new FileReader();
+    let selectedFile = e.target.files[0];
+
+    reader.onloadend = () => {
+      setFile(selectedFile);
+      setImagePreviewUrl(reader.result);
     };
 
-    axios
-      .post(`${backend}/api/posts`, postObject)
-      .then(document.location.reload());
+    reader.readAsDataURL(selectedFile);
+  };
+
+  const onSubmit = e => {
+    e.preventDefault();
+    if (file) {
+      let imageUpload = new FormData();
+      imageUpload.append("file", file);
+      axios.post(`${backend}/api/postimg`, imageUpload).then(response => {
+        let image_url = response.data.public_id;
+        if (game_id) {
+          var postObject = { image_url, message, user_id, game_id };
+        } else {
+          var postObject = { image_url, message, user_id };
+        }
+
+        axios
+          .post(`${backend}/api/posts`, postObject)
+          .then(() => document.location.reload());
```

Comment:
(I'd like someone to explain me why you need this, but you'll have to be convincing !)

Reply:
I was use to reload the page to be able to see the new post. But on other branch we implement redux and we will use it to re render the component instead of reloading the page. For the moment we have deleted the reloading

Reply:
I've change it using redux